app/uk/gov/hmrc/customs/declaration/services/declarationServices.scala

```diff
@@ -102,44 +98,32 @@ trait DeclarationService {
 
   private def callBackendAndNrs[A](implicit vpr: ValidatedPayloadRequest[A], hc: HeaderCarrier, sfId: SubscriptionFieldsId): Future[Either[Result, Option[NrSubmissionId]]] = {
 
```

Comment:
I think `val nrSubmissionId = ...` is confusing. May be clearer to use `vpr.conversationId.uuid` directly (you can do refactor inline variable in intellij) . This would be consistent with `send` in NRS service where we use `vpr.conversationId.toString` . So we can get probably remove case class `NrSubmissionId`